spec/system/admin/orders_spec.rb

```diff
@@ -122,6 +122,63 @@
       end
     end
 
+    context "bulk actions" do
+      context "resend confirmation email" do
+        it "can bulk send email to 2 orders" do
+          login_as_admin_and_visit spree.admin_orders_path
+
+          page.find("#listing_orders tbody tr:nth-child(1) input[name='order_ids[]']").click
+          page.find("#listing_orders tbody tr:nth-child(2) input[name='order_ids[]']").click
+
+          page.find("span.icon-reorder", text: "ACTIONS").click
+          within ".ofn-drop-down-with-prepend .menu" do
+            page.find("span", text: "Resend Confirmation").click
+          end
+
+          expect(page).to have_content "Are you sure you want to proceed?"
+
+          within ".reveal-modal" do
+            expect {
+              find_button("Confirm").click
+            }.to enqueue_job(ActionMailer::MailDeliveryJob).exactly(:twice)
+          end
+
+          expect(page).to have_content "Confirmation emails sent for 2 orders."
+        end
+
+        context "for a hub manager" do
+          before do
+            login_to_admin_as user
+            visit spree.admin_orders_path
+          end
+
+          it "cannnot send emails to orders if permission have been revoked in the meantime" do
+            page.find("#listing_orders tbody tr:nth-child(1) input[name='order_ids[]']").click
+
+            # Find the clicked order
+            order = Spree::Order.find_by(id: page.find("#listing_orders tbody tr:nth-child(1) input[name='order_ids[]']").value)
+            # Revoke permission for the current user on that specific order by changing its owners
+            order.update_attribute(:created_by, create(:user))
+            order.update_attribute(:distributor, create(:distributor_enterprise))
+            order.reload
```

Comment:
You don't need to reload the order here. Nothing else was changed in the database and we are not using the object later.
```suggestion
```